var/spack/repos/builtin/packages/libffi/package.py

```diff
@@ -32,4 +32,9 @@ def configure_args(self):
             # Spack adds its own target flags, so tell libffi not to
             # second-guess us
             args.append('--without-gcc-arch')
+        # At the moment, build scripts accept 'aarch64-apple-darwin'
+        # but not 'arm64-apple-darwin'.
+        # See: https://github.com/libffi/libffi/issues/571
+        if self.spec.satisfies('platform=darwin target=arm64'):
```

Comment:
This depends on #23817, but once that PR is in:
```suggestion
        if self.spec.satisfies('platform=darwin target=aarch64'):
```

Reply:
I'll merge as soon as this modification is in.